MyString.java

```diff
@@ -49,8 +85,21 @@ public static boolean subsetOf(String str1, String str2) {
      * @return a string consisting of the characters of str, separated by spaces.
      */
     public static String spacedString(String str) {
-        //// Replace the following statement with your code
-        return null;
+        String new_str="";
+        for(int i=0;i<str.length();i++)
+        {
+            if(i!=str.length()-1)
+            {
+                new_str+=str.charAt(i);
+                new_str+=" ";
+            }
+            else
+            {
+                new_str+=str.charAt(i);
+            }

```

Comment:
if you have same code in both if and else it usually means you can move it outside 
Consider the following
```
new_str+=str.charAt(i);
if(i != str.length()-1)  {
                new_str+=" ";
}

```